### PR TITLE
Regenerate IoT Hub control plane SDK to use new service API version

### PR DESCRIPTION
**Pull request by timtay-microsoft**
https://github.com/Azure/azure-rest-api-specs/pull/14894

Also adding a simple code generation script for local generation for other service API version increases later that we'll need to regenerate this SDK for

This new service API version is **not** deployed in every region yet, so this code is not ready to be released, but my team would like it checked in so the nightly built jars can be used for an upcoming bugbash.

### CODE REVIEW

File: sdk/iothub/generate.ps1 [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	--package-version=$newVersion `
+	--java `
+	--pipeline.modelerfour.additional-checks=false `
+	--pipeline.modelerfour.lenient-model-deduplication=true `
+	--azure-arm `
+	--verbose `
+	--sdk-integration `
+	--fluent=lite `
+	--java.fluent=lite `
+	--java.license-header=MICROSOFT_MIT_SMALL
```
Comment (weidongxu-microsoft) [inline sdk/iothub/generate.ps1:26]: You can try with one additional option `--generate-samples` to allow codegen to generate some sample codes for you (from swagger).

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/iothub/azure-resourcemanager-iothub/README.md [hunk 2]
```diff
@@ -74,13 +74,15 @@ See [API design][design] for general introduction on design and key concepts on
...
+
+
 ## Troubleshooting
 
 ## Next steps
 
 ## Contributing
 
-For details on contributing to this repository, see the [contributing guide](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).
+For details on contributing to this repository, see the [contributing guide](https://github.com/Azure/azure-sdk-for-java/blob/master/CONTRIBUTING.md).
```
Comment (weidongxu-microsoft) [inline sdk/iothub/azure-resourcemanager-iothub/README.md:85]: Please revert this line.

It is a bug in codegen, I will release 4.0.32 to fix it.

Comment (timtay-microsoft) [reply sdk/iothub/azure-resourcemanager-iothub/README.md:85]:
```suggestion
For details on contributing to this repository, see the [contributing guide](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).
```

File: sdk/iothub/generate.ps1 (hunk 1 above)
Comment (timtay-microsoft) [reply sdk/iothub/generate.ps1]: Good idea

File: sdk/iothub/azure-resourcemanager-iothub/README.md (hunk 2 above)
Comment (weidongxu-microsoft) [reply sdk/iothub/azure-resourcemanager-iothub/README.md]: 4.0.32 released

**Pull request merged**